### ISSUE TITLE
[Bug]: ERR_REQUIRE_ESM on @serwist/build

### ISSUE THREAD

**Issue by Zlkva** [serwist/serwist#34]
### Provide environment information

```
{
  "name": "roost",
  "version": "0.2.0",
  "private": true,
  "scripts": {
    "dev": "next dev",
    "build": "next build",
    "start": "next start",
    "lint": "next lint"
  },
  "dependencies": {
    "@fortawesome/fontawesome-svg-core": "^6.2.0",
    "@fortawesome/free-brands-svg-icons": "^6.2.0",
    "@fortawesome/free-regular-svg-icons": "^6.2.0",
    "@fortawesome/free-solid-svg-icons": "^6.2.0",
    "@fortawesome/react-fontawesome": "^0.2.0",
    "@serwist/next": "^8.4.0",
    "@serwist/precaching": "^8.4.0",
    "@serwist/sw": "^8.4.0",
    "isomorphic-unfetch": "^3.1.0",
    "js-cookie": "^3.0.1",
    "luxon": "^3.3.0",
    "moment": "^2.29.4",
    "next": "^14.0.4",
    "next-cookies": "^2.0.3",
    "normalize.css": "^8.0.1",
    "octokit": "^3.1.2",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "react-hot-toast": "^2.4.0",
    "react-image-gallery": "^1.2.11",
    "react-modal": "^3.16.1",
    "react-tabs": "^5.1.0",
    "react-twemoji": "^0.5.0",
    "rss-parser": "^3.13.0",
    "snoowrap": "^1.23.0"
  },
  "devDependencies": {
    "eslint": "^8.56.0",
    "eslint-config-next": "^14.0.4",
    "sass": "^1.49.10"
  }
}

```

### Which project is this issue for?

@serwist/build

### Link to reproduction - Issues with a link to complete (but minimal) reproduction code help us address them faster

https://projectisprivate-cantreplicate.com

### To reproduce

May be harder to reproduce. I had an existing NextJS project and I'm attempting to add serwist for PWA capabilities.

I followed the installation guide for NextJS on the documentation website for all steps.
https://serwist.pages.dev/docs/next/getting-started

I attempted to run the project afterwards with `npm run dev ` and received the error I'm seeing.

### Describe the bug

Upon running the project, I get the following error:
    
    $ npm run dev
    
    > roost@0.2.0 dev
    > next dev
    
    D:\code\roost\node_modules\@serwist\build\dist\index.cjs:65
        glob pattern strings.`,
                               ^
    
    Error [ERR_REQUIRE_ESM]: require() of ES Module D:\code\roost\node_modules\stringify-object\index.js from D:\code\roost\node_modules\@serwist\build\dist\index.cjs not supported.
    Instead change the require of index.js in D:\code\roost\node_modules\@serwist\build\dist\index.cjs to a dynamic import() which is available in all CommonJS modules.
        at mod.require (D:\code\roost\node_modules\next\dist\server\require-hook.js:65:28)
        at Object.<anonymous> (D:\code\roost\node_modules\@serwist\build\dist\index.cjs:13:23)
        at mod.require (D:\code\roost\node_modules\next\dist\server\require-hook.js:65:28)
        at Object.<anonymous> (D:\code\roost\node_modules\@serwist\webpack-plugin\dist\index.cjs:3:13)
        at mod.require (D:\code\roost\node_modules\next\dist\server\require-hook.js:65:28)
        at Object.<anonymous> (D:\code\roost\node_modules\@serwist\next\dist\index.cjs:7:21)
        at mod.require (D:\code\roost\node_modules\next\dist\server\require-hook.js:65:28)
        at Object.<anonymous> (D:\code\roost\next.config.js:4:21) {
      code: 'ERR_REQUIRE_ESM'
    }
    
    Node.js v18.17.0

### Expected behavior

The appropriate modules run, the project builds and is rendered in dev-mode.

**Comment by Zlkva** [comment]
Realistically I'm just looking for assistance with resolving the ERR_REQUIRE_ESM bug. Most answers I find are how to solve for it at the package maintainer level, but not at the "i am using your package" level.

**Comment by DuCanhGH** [comment]
Goddamn it not again...

**Comment by Zlkva** [comment]
> Goddamn it not again... We will need some tests this time.

if its absolutely needed, I can provide my exact project. It's just private on Github and I didn't want to go through all the steps of making it public and whatnot when this project is dreadfully simple in its function.

**Comment by DuCanhGH** [comment]
@Zlkva no, no need. It's just that `@serwist/build` was released with this error once before. So, basically, we will need some tests to ensure that this won't happen again.

**Comment by DuCanhGH** [comment]
@Zlkva should be fixed in 8.4.1. Give it a try!

**Comment by Zlkva** [comment]
> @Zlkva should be fixed in 8.4.1. Give it a try!

Bingo, no more issues on my side.

Thanks for the fast turnaround!
